src/extraction/special/refs/indirect.rs

```diff
@@ -0,0 +1,65 @@
+use winnow::{
+    ascii::multispace1,
+    combinator::{delimited, trace},
+    BStr, PResult, Parser,
+};
+
+use crate::extraction::{extract, Extract};
+
+use super::ReferenceId;
+
+/// The source for an indirect object, in the PDF body. It can be referenced in the PDF using a
+/// [`Reference`](super::Reference).
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Indirect<T> {
+    pub id: ReferenceId,
+    pub inner: T,
+}
+
+impl<'de, T> From<(ReferenceId, T)> for Indirect<T>
+where
+    T: Extract<'de>,
+{
+    fn from((id, inner): (ReferenceId, T)) -> Self {
+        Self { id, inner }
+    }
+}
+
+/// We go the extra mile and extract the trailing `endobj` tag. This is not actually needed,
+/// although it does serve as a kind of sanity check.
+impl<'de, T> Extract<'de> for Indirect<T>
+where
+    T: Extract<'de>,
+{
+    fn extract(input: &mut &'de BStr) -> PResult<Self> {
+        trace(
+            "livre-indirect",
+            (
+                extract,
+                delimited((b" obj", multispace1), extract, (multispace1, b"endobj")),
+            )
+                .map(Self::from),
+        )
+        .parse_next(input)
+    }
+}
```

Comment:
Note: we may need to make `Indirect` build rather than `Extract`... It really depends how we decide to implement `BuildFromRawDict`:

- using a `Build` struct directly, or
- through a generated, transient `Extract` type with some fields being `OptRef/Reference`.

The latter is definitely more involved, but would allow simple types to remain, well, simple.